src/main/app.js

```diff
@@ -353,10 +392,39 @@ fluid.defaults("gpii.app", {
             }
         },
         onGPIIReady: null,
+
         onAppReady: null,
+
+        // "onCreate" child component events that feed into onAppReady.
+        appZoomHandlerCreated: null,
```

Comment:
I don't think we need one event for each component, which is quite a scaling risk. All we need is a single listener which forwards from event onPSPPrerequisitesReady with priority "last" to a single new event

Reply:
This does not appear to be adequate.  There is already an onPSPReady event that does precisely that.  If I change [this line](https://github.com/the-t-in-rtf/gpii-app/blob/GPII-3744/tests/configs/gpii.tests.all.config.json5#L56) to use that event instead of onChildComponentsReady, I get new test errors when running the  first two integration tests.

Reply:
Could you paste the error transcripts? The fact that onPSPReady itself isn't enough suggests some further design problem

Reply:
Yes, I think this is the heart of what we need to address, and a transcript is already in your Skype from earlier.